Backend/ServerManager.py

```diff
@@ -39,21 +41,57 @@ def home():
 
 @app.route('/users', methods=['GET'])
 def api_tusers():
-    result = db.query("SELECT * FROM test_users;")
-    return jsonify(result)
+    if cache.should_update("users", 1):
+        result = db.query("SELECT * FROM test_users;")
+        users = []
+        for row in result.get_rows():
+            users.append({"id": row[0], "email": row[1]})
+        json = jsonify(users)
+        cache.add("users", json, 1)
+        return json
+    else:
+        print("Returning USERS results from cache")
+        return cache.retrieve("users")
 
 
 @app.route('/tweets', methods=['GET'])
 def api_ttweets():
-    result = db.query("SELECT * FROM test_tweets;")
-    return jsonify(result)
+    if cache.should_update("tweets", 1):
+        result = db.query("SELECT * FROM test_tweets;")
+        tweets = []
+        for row in result.get_rows():
+            tweets.append(DataStructures.Tweet(row[0], row[1], row[2], row[3]).__dict__)
```

Comment:
Optional*: my understanding is that Python uses underscores to indicate that a class member is 'protected/private', i.e. that it should preferably just be directly referenced inside the class. I'd argue that it'd be generally-better practice to wrap this __dict__ reference in a helper function of the Tweet class, like Tweet.toDict()

*if you think this is fussy or irrelevant, please let me know and I'll immediately resolve this review comment